src/cmd/services/m3coordinator/ingest/m3msg/ingest.go

```diff
@@ -223,7 +231,39 @@ func (op *ingestOp) resetWriteQuery() error {
 			Resolution:  op.sp.Resolution().Window,
 			Retention:   op.sp.Retention().Duration(),
 		},
-	})
+	}
+
+	if op.storeMetricsType {
+		var err error
+		wq.Annotation, err = op.convertTypeToAnnotation(op.metricType)
+		if err != nil {
+			return err
+		}
+	}
+
+	return op.q.Reset(wq)
+}
+
+func (op *ingestOp) convertTypeToAnnotation(tp ts.PromMetricType) ([]byte, error) {
+	if tp == ts.PromMetricTypeUnknown {
+		return nil, nil
+	}
+
+	atp, err := storage.PromMetricTypeToAnnotationPayloadType(tp)
+	if err != nil {
+		return nil, err
+	}
+
+	annotationPayload := annotation.Payload{MetricType: atp}
```

Comment:
This is missing the `HandleValueResets` field which should be set to `true` for counters (which indicates that some unit test should be added for such case).
I suggest to fix it by reverting the original `SeriesAttributesToAnnotationPayload` function and modifying to take `PromType` and `HandleValueResets` as arguments. Then you should be able to reuse it here directly, and avoid duplication of `annotation.Payload` construction code with all the consequences.